link/syscalls.go

```diff
@@ -80,11 +76,19 @@ var haveProgAttachReplace = internal.NewFeatureTest("BPF_PROG_ATTACH atomic repl
 	}
 
 	err = sys.ProgAttach(&attr)
-	if errors.Is(err, unix.EINVAL) {
-		return internal.ErrNotSupported
-	}
-	if errors.Is(err, unix.EBADF) {
-		return nil
+	if err != nil {
+
+		if errors.Is(err, unix.EINVAL) {
```

Comment:
Same nit as before, `errors.Is` doesn't have to be guarded by `if err != nil`.

Reply:
^^ Agreed that it's more noisy, & adds no extra value so will change that to it's previous state as mentioned above.